File: docs-examples/example-java/src/test/java/io/micronaut/quartz/docs/quickstart/MessageScheduler.java
package io.micronaut.quartz.docs.quickstart;

// tag::imports[]
import io.micronaut.quartz.annotation.QuartzKey;
import io.micronaut.quartz.annotation.QuartzSchedule;
import io.micronaut.quartz.annotation.ScheduleOn;
import org.quartz.JobKey;
// end:imports[]

// tag::clazz[]
@QuartzSchedule()
public interface MessageScheduler {
    @ScheduleOn(MessageJob.class)
    void sendMessage(@QuartzKey("message") String message);
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/quartz/docs/quickstart/MessageScheduler.kt
package io.micronaut.quartz.docs.quickstart

// tag::imports[]
import io.micronaut.quartz.annotation.QuartzKey
import io.micronaut.quartz.annotation.QuartzSchedule
import io.micronaut.quartz.annotation.ScheduleOn
import org.quartz.JobKey
// end:imports[]

// tag::clazz[]
@QuartzSchedule()
interface MessageScheduler {
    @ScheduleOn(MessageJob::class)
    fun sendMessage(key: JobKey, @QuartzKey("message") message: String)
}
// end::clazz[]
